pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerConfig.java

```diff
@@ -143,7 +143,7 @@ public static WorkerConfig load(String yamlFile) throws IOException {
 
     public String getWorkerId() {
         if (StringUtils.isBlank(this.workerId)) {
-            this.workerId = getWorkerHostname();
+            this.workerId = String.format("%s-%s", this.getWorkerHostname(), this.getWorkerPort());
```

Comment:
This change has implications for other places where both workerId and workerport are being used. Maybe add a seperate function?

Reply:
it was a bug actually, if multiple worker instances start on the same host but different port then it doesn't create unique workerId and it was resulting into incorrect assignment. Therefore, it is part of bug-fix also.